qiskit/pulse/model/__init__.py

```diff
@@ -0,0 +1,83 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+.. _pulse-logical-elements-frames:
+
+=======================================
+Logical Elements & Frames (:mod:`qiskit.pulse.logical_elements_frames`)
+=======================================
+
+Pulse is meant to be agnostic to the underlying hardware implementation, while still allowing
+low-level control. Qiskit Pulse's logical element and frames are meant to create a flexible framework
+to define where pulses instructions are applied, and what would be their carrier frequency and phase
+(because typically AC pulses are used). Each :class:`LogicalElement` represents a separate component
+in the quantum system on which instructions could be applied. On the other hand, each :class:`Frame`
+represents a frequency and phase duo for the carrier of the pulse.
```

Comment:
We might want to be more careful here. In the `Frame` class, you specify initial frequency and initial phase. The `Frame` also represents the object to reference to change the frequency and phase duo over time by applying frequency and phase instructions but that frequency and phase duo can only be determined by scheduling a pulse program.

Reply:
I agree, this is a delicate topic.
My thinking was that we need to set the initial frequency and phase at time `t=0` for `Frame`s that don't have a default value taken from the backend. If you think of a pulse program (like the old `Schedule`) this is sufficient. Things might get trickier if we think of a program with several pulse gates. In principle the initial values are still enough, but you will have to track the phases across the entire program? (To be fair, this is not handled well even with the current stack)

Reply:
Yes, this was something else I was thinking about how to ask about. One of the motivations for this work on pulse frames was to help with the qutrit and cross-resonance cases where there is a need to apply a shift and have it persist across pulse gates. Do we try to capture that here? Or is that left to the backend implementation? At some point, we need to introduce a way to change the IBM backend implementation so that a pulse gate can influence later pulse gates.

Reply:
That's a great question, which we haven't ironed out yet.

I think the user friendly approach would be to define the initial frequency and phase at `t=0` where `t=0` is not the beginning of the pulse gate, but rather the beginning of the circuit. This gives a concrete meaning to the frame which could then be tracked over time. The job of actually tracking such a frame would have to be assigned to either the backend or the frontend. However, this introduces a global dependency to the pulse gates which is not currently available in the stack. Perhaps this global dependency is needed anyway if we want to carry out phases and frequencies correctly across pulse gates? If that's the case, we can sort all of these issues via a compiler run which will schedule the circuit as a whole, and address the frames on a global basis.

Reply:
I think eventually the management of frame is delegated to the backend. Frame is a _global static variable_ and shared among pulse gates, and local shift/set frequency instruction doesn't update frame value. 

>  Do we try to capture that here? Or is that left to the backend implementation?

Qiskit doesn't define how things must be implemented on the physical layer, so it's just a syntax. Actual implementation must be done on the backend, still IBM provider can provider a plug-in pass that computes local frame shift in each pulse gate (provider pulg-in pass should know backend implementation). If you run the same program on Qiskit Dynamics, phase and frequency are already carried over because it schedules entire circuit before running simulation. This is why I prefer pass based architecture.